packages/chat-widget/src/index.tsx

```diff
@@ -352,6 +357,75 @@ const isInputDisabled = (responses: Response[]): boolean => {
   return new URLSearchParams(payload).get("nlx:input-disabled") === "true";
 };
 
+const findActiveUpload = (responses: Response[]): UploadUrl | null => {
+  const lastBotResponse = findLast(
```

Comment:
Is there anyway there can be a partial upload and in the meantime another response comes in?

Reply:
Yes this is theoretically possible but I would say unlikely. Should probably change the state management to keep track of things a bit more granular, but I'm scared to do it now.

Reply:
Maybe then at least add a TODO comment explaining this.

Reply:
Added in 0f99847b43e8